test/test_mpc.py

```diff
@@ -59,10 +59,8 @@ def _check(self, encrypted_tensor, reference, msg, dst=None, tolerance=None):
             logging.info("Result - Reference = %s" % (tensor - reference))
         self.assertTrue(test_passed, msg=msg)
         self.assertTrue(hasattr(encrypted_tensor, "_mac"), "_mac attribute failed to propagate")
-        self.assertEqual(encrypted_tensor._mac, 0, "_mac attribute has incorrect value")
-
-        self.assertTrue(hasattr(encrypted_tensor, "_mac"), "_mac attribute failed to propagate")
-        self.assertEqual(encrypted_tensor._mac, 0, "_mac attribute has incorrect value")
+        if dst is None:
+           self.assertTrue(encrypted_tensor._tensor.reveal().mul(MPCTensor.alpha.reveal()).eq(encrypted_tensor._mac.reveal()).all(), msg="Mac Arithmetic Mismatch")
```

Comment:
This is a good start. I think there is a little more we should do before closing this.

Here, we are only testing that the mac is correct when `dst is None`. In order to test this properly, we need to test both when `dst is None` and when `dst` takes a value.

So, we should add some `elif` blocks here for the following:

Case 1: Current process is not the `dst` party. Here we want the result to be None.

Case 2: Current process is the `dst` party. Here we want the test to be the same as above.



```
elif dst != self.rank:
    self.assertIsNone(encrypted_tensor._mac.reveal(dst=dst)
else:
    # same as above code.
```

In order to avoid copy-pasting code, these can be consolidated:

```
if dst is None or dst == self.rank:
    # Above code with reveal(dst=dst) for each reveal
elif dst is not None:
    self.assertIsNone(encrypted_tensor._mac.reveal(dst=dst))
```

Let me know if you have questions.

Reply:
One quick question, what exactly  `.rank` attribute do?

Reply:
Also, after I modified test to this:
```
if dst is None or dst == self.rank:
    self.assertTrue(encrypted_tensor._tensor.reveal().mul(MPCTensor.alpha.reveal()).eq(encrypted_tensor._mac.reveal()).all(), msg="Mac Arithmetic Mismatch")
elif dst is not None:
    self.assertIsNone(encrypted_tensor._mac.reveal(dst=dst))
```

It hanged again on `self.assertTrue(encrypted_tensor._tensor.reveal().mul(MPCTensor.alpha.reveal()).eq(encrypted_tensor._mac.reveal()).all(), msg="Mac Arithmetic Mismatch")` for some reason.

Reply:
It seems to be stuck only when dst == self.rank